### ISSUE TITLE
🐛 biome 2.4.7 breaks speculation rules

### ISSUE THREAD

**Issue by Th3S4mur41** [biomejs/biome#9506]
### Environment information

<details>

```bash

CLI:
  Version:                      2.4.7
  Color support:                true

Platform:
  CPU Architecture:             x86_64
  OS:                           linux

Environment:
  BIOME_LOG_PATH:                    unset
  BIOME_LOG_PREFIX_NAME:             unset
  BIOME_LOG_LEVEL:                   unset
  BIOME_LOG_KIND:                    unset
  BIOME_CONFIG_PATH:                 unset
  BIOME_THREADS:                     unset
  BIOME_WATCHER_KIND:                unset
  BIOME_WATCHER_POLLING_INTERVAL:    unset
  NO_COLOR:                     unset
  TERM:                         xterm-256color
  JS_RUNTIME_VERSION:           v24.13.1
  JS_RUNTIME_NAME:              node
  NODE_PACKAGE_MANAGER:         npm/11.8.0

Biome Configuration:
  Status:                       Loaded successfully
  Path:                         biome.json
  Formatter enabled:            true
  Linter enabled:               true
  Assist enabled:               true
  VCS enabled:                  true
  HTML full support enabled:    true

```

</details>

### What happened?

1. Bump @biomejs/biome from 2.4.6 to 2.4.7
2. Biome format removes curly bracket in [speculation rules](https://developer.mozilla.org/en-US/docs/Web/API/Speculation_Rules_API)
3. Parser fails with "Expected a semicolon or an implicit semicolon after a statement, but found none"

https://biomejs.dev/playground/?code=PAAhAGQAbwBjAHQAeQBwAGUAIABoAHQAbQBsAD4ACgA8AGgAdABtAGwAIABsAGEAbgBnAD0AIgBlAG4AIgA%2BAAoAIAAgADwAaABlAGEAZAA%2BAAoAIAAgADwAcwBjAHIAaQBwAHQAIAB0AHkAcABlAD0AIgBzAHAAZQBjAHUAbABhAHQAaQBvAG4AcgB1AGwAZQBzACIAPgAKACAAIAAgACAAewAKACAAIAAgACAAIAAgACIAcAByAGUAcgBlAG4AZABlAHIAIgA6AAoAIAAgACAAIAAgACAAWwAKACAAIAAgACAAIAAgACAAIAB7AAoAIAAgACAAIAAgACAAIAAgACAAIAAiAHcAaABlAHIAZQAiADoAewAiAGgAcgBlAGYAXwBtAGEAdABjAGgAZQBzACIAOgAiAC8AKgAiAH0ALAAKACAAIAAgACAAIAAgACAAIAAgACAAIgBlAGEAZwBlAHIAbgBlAHMAcwAiADoAIgBtAG8AZABlAHIAYQB0AGUAIgAKACAAIAAgACAAIAAgACAAIAB9AAoAIAAgACAAIAAgACAAXQAsAAoAIAAgACAAIAAgACIAcAByAGUAZgBlAHQAYwBoACIAOgAKACAAIAAgACAAIAAgACAAWwAKACAAIAAgACAAIAAgACAAIAAgAHsACgAgACAAIAAgACAAIAAgACAAIAAgACAAIgB3AGgAZQByAGUAIgA6AHsAIgBoAHIAZQBmAF8AbQBhAHQAYwBoAGUAcwAiADoAIgAvACoAIgB9ACwACgAgACAAIAAgACAAIAAgACAAIAAgACAAIgBlAGEAZwBlAHIAbgBlAHMAcwAiADoAIgBtAG8AZABlAHIAYQB0AGUAIgAKACAAIAAgACAAIAAgACAAIAAgAH0ACgAgACAAIAAgACAAIAAgAF0ACgAgACAAIAAgAH0ACgAgACAAPAAvAHMAYwByAGkAcAB0AD4ACgA8AC8AaABlAGEAZAA%2BAAoAPABiAG8AZAB5AD4ACgAgACAACgA8AC8AYgBvAGQAeQA%2BAAoAPAAvAGgAdABtAGwAPgAKAA%3D%3D&language=html

### Expected result

Parser should not fail and format shouldn't been changed to invalid

### Code of Conduct

- [x] I agree to follow Biome's Code of Conduct

**Comment by ematipico** [comment]
It should be easy to fix. We should just need to add a new detector here

https://github.com/biomejs/biome/blob/c31197463b4ed4a8eaeabb37b86531765bb22967/crates/biome_service/src/embed/registry.rs#L39
